src/cpp/ext/otel/otel_plugin.cc

```diff
@@ -175,47 +176,61 @@ void OpenTelemetryPluginBuilder::BuildAndRegisterGlobal() {
   }
   if (metrics_.contains(
           OTelClientAttemptSentTotalCompressedMessageSizeInstrumentName())) {
-    g_otel_plugin_state_->client.attempt.sent_total_compressed_message_size =
-        meter->CreateUInt64Histogram(std::string(
-            OTelClientAttemptSentTotalCompressedMessageSizeInstrumentName()));
+    g_otel_plugin_state_->client.attempt
+        .sent_total_compressed_message_size = meter->CreateUInt64Histogram(
+        std::string(
+            OTelClientAttemptSentTotalCompressedMessageSizeInstrumentName()),
+        "Compressed message bytes sent per call attempt", "By");
   }
   if (metrics_.contains(
           OTelClientAttemptRcvdTotalCompressedMessageSizeInstrumentName())) {
-    g_otel_plugin_state_->client.attempt.rcvd_total_compressed_message_size =
-        meter->CreateUInt64Histogram(std::string(
-            OTelClientAttemptRcvdTotalCompressedMessageSizeInstrumentName()));
+    g_otel_plugin_state_->client.attempt
+        .rcvd_total_compressed_message_size = meter->CreateUInt64Histogram(
+        std::string(
+            OTelClientAttemptRcvdTotalCompressedMessageSizeInstrumentName()),
+        "Compressed message bytes received per call attempt", "By");
   }
   if (metrics_.contains(OTelServerCallStartedInstrumentName())) {
     g_otel_plugin_state_->server.call.started = meter->CreateUInt64Counter(
-        std::string(OTelServerCallStartedInstrumentName()));
+        std::string(OTelServerCallStartedInstrumentName()),
+        "Number of server calls started", "{call}");
   }
   if (metrics_.contains(OTelServerCallDurationInstrumentName())) {
     g_otel_plugin_state_->server.call.duration = meter->CreateDoubleHistogram(
-        std::string(OTelServerCallDurationInstrumentName()));
+        std::string(OTelServerCallDurationInstrumentName()),
+        "End-to-end time taken to complete a call from server transport's "
+        "perspective",
+        "s");
   }
   if (metrics_.contains(
           OTelServerCallSentTotalCompressedMessageSizeInstrumentName())) {
     g_otel_plugin_state_->server.call.sent_total_compressed_message_size =
-        meter->CreateUInt64Histogram(std::string(
-            OTelServerCallSentTotalCompressedMessageSizeInstrumentName()));
+        meter->CreateUInt64Histogram(
+            std::string(
+                OTelServerCallSentTotalCompressedMessageSizeInstrumentName()),
+            "Compressed message bytes sent per server call", "By");
   }
   if (metrics_.contains(
           OTelServerCallRcvdTotalCompressedMessageSizeInstrumentName())) {
     g_otel_plugin_state_->server.call.rcvd_total_compressed_message_size =
-        meter->CreateUInt64Histogram(std::string(
-            OTelServerCallRcvdTotalCompressedMessageSizeInstrumentName()));
+        meter->CreateUInt64Histogram(
+            std::string(
+                OTelServerCallRcvdTotalCompressedMessageSizeInstrumentName()),
+            "Compressed message bytes sent per server call", "By");
```

Comment:
s/sent/received

Reply:
Thanks! I'm so glad I asked you to take a look :D